skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,417 @@
+
+        self.assertTrue(f._interval_metadata is not None)
+        self.assertListEqual(f.intervals, [(1, 2), (4, 7)])
+        self.assertListEqual(f.boundaries, [(True, False), (False, False)])
+        self.assertDictEqual(f.metadata, {'name': 'sagA',
+                                          'function': 'transport'})
+
+    def test_bad_constructor(self):
+        with self.assertRaises(ValueError):
+            Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[1, (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+    def test_repr(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp1 = (r"Interval(intervals=[(1, 2), (4, 7)], "
+                "metadata={'name': 'sagA', 'function': 'transport'})")
+        res = repr(f)
+        # because dictionaries are random
+        self.assertTrue(res, exp1)
+
+    def test_getitem(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertEqual(f['name'], 'sagA')
+        self.assertEqual(f['function'], 'transport')
+
+    def test_setitem(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f['name'] = 'sagB'
+        self.assertEqual(f['name'], 'sagB')
+        self.assertEqual(f['function'], 'transport')
+
+    def test_cmp(self):
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        f2 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(10, 20)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertTrue(f1 < f2)
+        self.assertTrue(f1 <= f1)
+        self.assertTrue(f1 <= f2)
+        self.assertFalse(f1 > f2)
+        self.assertFalse(f1 >= f2)
+        self.assertTrue(f1 >= f1)
+
+    def test_contains(self):
+        f1 = Interval(interval_metadata=IntervalMetadata(),
+                      intervals=[(1, 2)],
+                      boundaries=[(True, False), (False, False)],
+                      metadata={'name': 'sagA', 'function': 'transport'})
+        self.assertTrue('name' in f1)
```

Comment:
Is that really part of the unittest framework?  Looking around it looks like that is only included in [Django](https://groups.google.com/forum/#!topic/django-users/jkRRgPUAhoM).